Add wizard-like export view for workflow invocations

davelopez opened the pull request:
This is an attempt to make the workflow invocation export process a bit more guided and user-friendly. It also explores using wizard-like interfaces for "complex" processes with many options.

![InvocationWizardFinal](https://github.com/user-attachments/assets/159124a8-27fa-4f2c-b731-d48cdd0b28b3)


If we like this approach we can make it more generic and incorporate it, for example in the history export (and archival process) or in other places where the user has to decide between multiple options, like collection building, etc.

---

## Generic Wizard Usage
I've extracted the basic logic and the base component into `GenericWizard`. Here is a basic example of how to use it.

**MyWizard.vue**
```vue
<script setup lang="ts">
import { reactive, ref } from "vue";

import { useWizard } from "@/components/Common/Wizard/useWizard";

import GenericWizard from "@/components/Common/Wizard/GenericWizard.vue";

// This reactive object holds the data for the entire wizard.
// Use the values in the wizard steps to control the wizard flow
// and to validate the data before submitting.
const wizardData = reactive({
    enableHiddenStep: false,
    confirmDone: false,
});

// Use the useWizard composable to define the wizard steps
const wizard = useWizard({
    intro: {
        label: "Initial Step",
        instructions: "First instructions go here",
        isValid: () => true,
        isSkippable: () => false,
    },
    "optional-step": {
        label: "Optional",
        instructions: "The instructions for the optional step",
        isValid: () => true,
        isSkippable: () => !wizardData.enableHiddenStep,
    },
    end: {
        label: "Final step",
        instructions: "Mark the checkbox to continue",
        isValid: () => wizardData.confirmDone,
        isSkippable: () => false,
    },
});

const submitButtonLabel = ref("Done");
const isBusy = ref(false);

// This function is called when the wizard is completed
async function onWizardSubmit() {
    isBusy.value = true;
    console.log("Wizard submit...", wizardData);
    await new Promise((resolve) => setTimeout(resolve, 2000));
    console.log("Wizard submit DONE");
    isBusy.value = false;
}
</script>

<template>
    <div>
        <GenericWizard
            title="My Simple Wizard"
            :use="wizard"
            :is-busy="isBusy"
            :submit-button-label="submitButtonLabel"
            @submit="onWizardSubmit">
            <!-- To define the content of each step, use the `v-if` directive with 
            the `wizard.isCurrent` method and pass the corresponding step key -->
            <div v-if="wizard.isCurrent('intro')">
                <p>This is the content of the initial step</p>
                <BFormCheckbox v-model="wizardData.enableHiddenStep" switch> Enable Optional Step </BFormCheckbox>
            </div>

            <div v-if="wizard.isCurrent('optional-step')">
                <p>This is the content of the optional step</p>
                <b>Nothing to do here...</b>
            </div>

            <div v-if="wizard.isCurrent('end')">
                <BFormCheckbox v-model="wizardData.confirmDone">
                    Enable the <b>{{ submitButtonLabel }}</b> button
                </BFormCheckbox>
            </div>
        </GenericWizard>
    </div>
</template>
```

![MyWizard](https://github.com/user-attachments/assets/2c3ac80a-9426-4595-a34d-fae0f283adc0)


## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

davelopez committed:
Move RDMCredentialsInfo component

And remove unused prop

davelopez committed:
Add onDismiss event to PersistentTaskProgressMonitorAlert

davelopez committed:
Refactor move interfaces

davelopez committed:
Add optional image logos to invocation export plugins

davelopez committed:
Add getStoredProgressData to progress monitor utilities

davelopez committed:
Refactor extract RDMDestinationSelector from ExportRDMForm

So it can be used in other contexts as well.

davelopez committed:
Replace invocation export plugins view with wizard

davelopez committed:
Remove unused test file and component

davelopez committed:
Fix selectors in tests for PersistentTaskProgressMonitorAlert

davelopez committed:
Make wizard selection cards more visible

davelopez committed:
Improve export Summary in InvocationExportWizard.vue

davelopez committed:
Add FileSourceNameSpan component

Displays the name of a remote source by providing the URI

davelopez committed:
Display more information on remote export progress result

davelopez committed:
Improve export destination summary in InvocationExportWizard

davelopez committed:
Fix getFileSourceByUri function in fileSources composable

Make sure it also matches file sources without ID (stock file sources like FTP, etc.)

davelopez committed:
Refactor remove unnecessary onMounted hook

davelopez committed:
Refactor usePersistentProgressTaskMonitor to use isDataExpired helper function

davelopez committed:
Disable buttons when the wizard is busy

davelopez committed:
Include a "Learn more" link on invocation export plugins

davelopez committed:
Refactor export wizard summary

To conditionally show file name input and include data checkbox

davelopez committed:
Remove additionalActions from invocation export plugins

It will be handled now by the wizard logic

davelopez committed:
Refactor BioCompute send to BCODB

Extract logic into a service module and Integrate it into the wizard process.

davelopez committed:
Disable wizard while sending to BCODB

davelopez committed:
Refactor extract GenericWizard component and useWizard composable

davelopez committed:
Refactor InvocationExportWizard to use GenericWizard

davelopez committed:
Display missing step bodies is GenericWizard

The body default value will remind you to define an appropriate body for the step.

davelopez committed:
Fix condition for skipping bcodb step

itisAliRH commented on client/src/components/Workflow/Invocation/Export/ExistingInvocationExportProgressCard.vue:
```diff
@@ -0,0 +1,105 @@
+</script>
+
+<template>
+    <BCard v-show="hasExistingStsExportData || hasExistingRemoteExportData">
```
Minor improvment:
```suggestion
    <BCard v-show="hasExistingStsExportData || hasExistingRemoteExportData" class="mb-2">
```
|Before|After|
|---|----|
|![Screenshot From 2024-09-17 15-13-18](https://github.com/user-attachments/assets/626482ad-b657-4927-920c-8adc2ae6f6c6)|![Screenshot From 2024-09-17 15-12-58](https://github.com/user-attachments/assets/45404021-f576-4d76-a302-e94780bf2666)|

davelopez committed:
Add margin to ExistingInvocationExportProgressCard

Co-authored-by: Alireza Heidari <itisalirh@gmail.com>

davelopez committed:
Handle export format change

If we go back and change the export format after selecting some of the other options with a the previous format, we should ensure the options are compatible with that new format.

davelopez committed:
Change RO-Crate URL link to WF Run Crate link
